jobs/etl_jobs/external/metabase/query.py

```diff
@@ -0,0 +1,124 @@
+        if "joins" in self.card_info["dataset_query"]["query"]:
+            for cond in self.card_info["dataset_query"]["query"]["joins"]:
+                if cond["source-table"] == legacy_table_id:
+                    cond["source-table"] = new_table_id
+                for field in cond["condition"]:
+                    if isinstance(field, list):
+                        if field[0] == "field":
+                            field[1] = mapped_fields_dict.get(field[1], field[1])
+                        else:
+                            for el in field:
+                                if el[0] == "field":
+                                    el[1] = mapped_fields_dict.get(el[1], el[1])
+        if "order-by" in self.card_info["dataset_query"]["query"]:
+            for order in self.card_info["dataset_query"]["query"]["order-by"]:
+                order[1][1] = mapped_fields_dict.get(order[1][1], order[1][1])
+
+        if "expressions" in self.card_info["dataset_query"]["query"]:
+            for key, value in self.card_info["dataset_query"]["query"][
+                "expressions"
+            ].items():
+                stack = [value]
+                while stack:
+                    current = stack.pop()
+                    for item in current:
+                        if isinstance(item, list):
+                            if item and item[0] == "field" and len(item) > 1:
+                                item[1] = mapped_fields_dict.get(item[1], item[1])
+                            else:
+                                stack.append(item)
+
+    def update_viz_settings(self, mapped_fields_dict):
+        if "table.columns" in self.card_info["visualization_settings"]:
+            for column in self.card_info["visualization_settings"]["table.columns"]:
+                if "field_ref" in column and column.get("field_ref")[0] == "field":
+                    column["field_ref"][1] = mapped_fields_dict.get(
+                        column["field_ref"][1], column["field_ref"][1]
+                    )
+                elif "fieldRef" in column and column.get("fieldRef")[0] == "field":
+                    column["fieldRef"][1] = mapped_fields_dict.get(
+                        column["fieldRef"][1], column["fieldRef"][1]
+                    )
+
+    def update_result_metadata(self, mapped_fields_dict):
+
+        if self.card_info["result_metadata"]:
+            for result in self.card_info["result_metadata"]:
+                if "field_ref" in result and result["field_ref"][0] == "field":
+                    if isinstance(result["field_ref"][1], int):
+                        result["field_ref"][1] = mapped_fields_dict.get(
+                            result["field_ref"][1], result["field_ref"][1]
+                        )
+                        result["id"] = mapped_fields_dict.get(
+                            result["id"], result["id"]
+                        )
```

Comment:
```suggestion
                if "field_ref" in result and result["field_ref"][0] == "field" and isinstance(result["field_ref"][1], int):
                      result["field_ref"][1] = mapped_fields_dict.get(
                          result["field_ref"][1], result["field_ref"][1]
                      )
                      result["id"] = mapped_fields_dict.get(
                          result["id"], result["id"]
                      )
```
C'est du détail et c'est pas un indent mais des espaces qu'il y a dans cette suggestion de code